### PR TITLE
rg35xx garlicos 1.3.5 compatibility and build

### PR DESCRIPTION
**Pull request by JoeStaff**
Introduces rg35xx GarlicOS v1.3.5 build and a few extra joy buttons to allow use of volume buttons, menu button, and "triggers"

### CODE REVIEW

**Comment by AJenbo** [comment]
Well that was short lived, did you intend to close it?

File: Packaging/rg35xx_garlic/skeleton_GarlicOS/ROMS/PORTS/Diablo/.local/share/diasurgical/devilution/diablo.ini [hunk 1]
```diff
@@ -0,0 +1,221 @@
+[Language]
```
Comment (glebm) [inline Packaging/rg35xx_garlic/skeleton_GarlicOS/ROMS/PORTS/Diablo/.local/share/diasurgical/devilution/diablo.ini:1]: This is going to get out of sync very quickly. Why is this file needed?
Our general approach for platforms that need different defaults is to set the defaults via CMake variables => `#define`.

File: Source/controls/controller_buttons.cpp [hunk 2]
```diff
@@ -260,6 +260,62 @@ string_view ToGenericButtonText(ControllerButton button)
...
+	case devilution::ControllerButton_NONE:
+		return "None";
+	case devilution::ControllerButton_IGNORE:
+		return "Ignored";
+	default:
+		return "Unknown";
+	}
+}
+
+string_view ToGenericButtonText2(ControllerButton button)
```
Comment (glebm) [inline Source/controls/controller_buttons.cpp:286]: What's this for? RG35xx uses the Nintendo button layout.
Simply `set(DEVILUTIONX_GAMEPAD_TYPE Nintendo)` in `rg35xx_garlic.cmake`.

File: Source/controls/devices/joystick.cpp [hunk 3]
```diff
@@ -65,6 +70,20 @@ StaticVector<ControllerButtonEvent, 4> Joystick::ToControllerButtonEvents(const
 		case JOY_BUTTON_BACK:
 			return { ControllerButtonEvent { ControllerButton_BUTTON_BACK, up } };
 #endif
+#ifdef JOY_BUTTON_MENU
+		case JOY_BUTTON_MENU:
+			return { ControllerButtonEvent { ControllerButton_BUTTON_MENU, up } };
+#endif
+#ifdef JOY_BUTTON_VOLUME_UP
+		case JOY_BUTTON_VOLUME_UP:
+			if(up)increasevolume();
```
Comment (glebm) [inline Source/controls/devices/joystick.cpp:79]: Aren't the volume buttons used to adjust system volume by default like on other handhelds?
Anyway, there should be no event handling in here, joystick mapping should only translate joystick events to ControllerButtons.

File: Source/controls/devices/joystick.cpp [hunk 4]
```diff
@@ -94,6 +113,33 @@ StaticVector<ControllerButtonEvent, 4> Joystick::ToControllerButtonEvents(const
 		return joystick->GetHatEvents();
 	}
 	case SDL_JOYAXISMOTION:
+	{
+		printf("Axis Motion: %d = value: %d\n",event.jaxis.axis,event.jaxis.value);
```
Comment (glebm) [inline Source/controls/devices/joystick.cpp:117]: `printf` left over from debugging.
